lib/oscap_tests.pm

```diff
@@ -123,186 +711,541 @@ sub upload_logs_reports {
+            my @exclusions = ();
+            $_[0] = \@exclusions;
+        }
+        return $found;
+    }
+}
 
+sub oscap_security_guide_setup {
+    # Main test setup function
     select_console 'root-console';
+    # Setting $full_ansible_file_path aftr got ansible_profile_ID from test
+    $full_ansible_file_path = $ansible_file_path . $ansible_profile_ID;
 
+    # Refresh repositories
+    zypper_call('ref -s', timeout => 180);
     # Install packages
     zypper_call('in openscap-utils scap-security-guide', timeout => 180);
-
     # Record the pkgs' version for reference
     my $out = script_output("zypper se -s openscap-utils scap-security-guide");
     record_info("Pkg_ver", "openscap security guide packages' version:\n $out");
-
     # Set ds file
     set_ds_file();
 
     # Check the ds file information for reference
-    my $f_ssg_ds = is_sle ? $f_ssg_sle_ds : $f_ssg_tw_ds;
-    $out = script_output("oscap info $f_ssg_ds");
+    $f_ssg_ds = is_sle ? $f_ssg_sle_ds : $f_ssg_tw_ds;
+    $out = script_output("oscap info $f_ssg_ds", quiet => 1);
     record_info("oscap info", "\"# oscap info $f_ssg_ds\" returns:\n $out");
-
     # Check the oscap version information for reference
     $out = script_output("oscap -V");
     record_info("oscap version", "\"# oscap -V\" returns:\n $out");
+
+    # Get the tests configuration file from repository and set global configuration varables
+    get_tests_config();
+    push(@test_run_report, "[configuration]");
+    $out = script_output("date", quiet => 1);
+    push(@test_run_report, "date = $out");
+    push(@test_run_report, "profile_ID = $profile_ID");
+    push(@test_run_report, "ansible_profile_file_name = $ansible_profile_ID");
+    push(@test_run_report, "use_content_type = $use_content_type");
+    push(@test_run_report, "remove_rules_missing_fixes = $remove_rules_missing_fixes");
+    push(@test_run_report, "use_exclusions = $use_exclusions");
+    push(@test_run_report, "evaluate_count = $evaluate_count");
+
+    # Replace original ds and xccdf files whith downloaded from local repository
+    set_ds_file_name();
+    push(@test_run_report, "sle_version = $sle_version");
+    push(@test_run_report, "ssg_sle_ds = $ssg_sle_ds");
+    push(@test_run_report, "ssg_sle_xccdf = $ssg_sle_xccdf");
+    my $arch = get_var 'ARCH';
+    push(@test_run_report, "os_arch = $arch");
+    my $type;
+    if ($ansible_remediation == 1) {
+        $type = 'ansible';
+    }
+    else {
+        $type = 'bash';
+    }
+    push(@test_run_report, "remediation_type = $type");
+    my $build_url = get_var 'CASEDIR';
+    push(@test_run_report, "build_url = $build_url");
+    my $iso_name = get_var 'ISO';
+    push(@test_run_report, "iso_name = $iso_name");
+    my $full_name = get_var 'NAME';
+    push(@test_run_report, "full_name = $full_name");
+    my $schedule = get_var 'YAML_SCHEDULE';
+    push(@test_run_report, "schedule = $schedule");
```

Comment:
all this block could be extracted to a function called for example `create_run_report`

Reply:
The reason why I did not create create_run_report function - data collection done in multiple functions and in different ways, code never reused.